### ISSUE TITLE
can I transfer your caffe model to a tensorflow model?

### ISSUE THREAD

**Issue by hx173149** [facebook/C3D#137]
hello, I have used your caffe code to do my work, it has a fantastic effort, recently I want to transfer this net to a tensorflow code, but I got 10% accuracy decrease ,finally I think the cause maybe in your caffe code I have used your pre-trained model: conv3d_deepnetA_sport1m_iter_1900000(this dataset is so huge!!!), but I can't use this model immediately. So I think the most simple way is transfer this model to a tensorflow model......
thanks ,do you have any better suggestion? thank you~

**Comment by dutran** [comment]
If you are working on tensorflow, then converting c3d model to tesorflow model seems a good option.

**Comment by hx173149** [comment]
thank you

**Comment by hx173149** [comment]
hello, I have converting the C3D code to a tensorflow version (in UCF101 dataset), the source code in [my repository](https://github.com/hx173149/C3D-tensorflow) (https://github.com/hx173149/C3D-tensorflow),if you think I have made some mistakes or you have some advices to me, please contact me, thanks~

**Comment by toddnguyen** [comment]
Hello, were you ever able to transfer the pretrained model for sport1m for use on tensorflow?
Thanks

**Comment by hx173149** [comment]
@toddnguyen I have transferred the sports1M model from caffe version to TF version, you can download the relevant code from here:
https://github.com/hx173149/C3D-tensorflow
and the pretrained TF model from here:
https://www.dropbox.com/sh/8wcjrcadx4r31ux/AAAkz3dQ706pPO8ZavrztRCca?dl=0

**Comment by dutran** [comment]
@hx173149 Thanks for your good contribution. I might be very useful for Tensorflow user.

**Comment by chuckcho** [comment]
I have instructions and scripts for converting sport1mil C3D-caffe model into Tensorflow and testing with a sample video: https://github.com/chuckcho/c3d-keras

**Comment by dutran** [comment]
@chuckcho thanks a lot for your great contribution.

**Comment by djour** [comment]
@chuckcho nice work!

**Comment by bityangke** [comment]
I try to convert ResNet C3D (C3D 1.1 version) to keras using the same code as I convert C3D 1.0 version.
But network output a category（57） no matter what inputs.
Can you give me some suggestions？
@hx173149 @chuckcho @dutran 
1. get the caffemodel to h5 file
`import caffe 
import numpy as np
import h5py

model_file = 'ResNet-C3D.prototxt' 
pretrained_model = 'ResNet-C3D-UCF101.caffemodel'

net = caffe.Net(model_file,pretrained_model,caffe.TEST)

f = h5py.File('resnet50.h5','w')

for name in net.params.keys():
    if len(net.params[name])==1:
        print "%s has only one params"%name
        grp = f.create_group(name)
        grp.create_dataset('weights',data=net.params[name][0].data)
        grp.create_dataset('bias',data=np.zeros(shape=(net.params[name][0].data.shape[0],)))
    else:
        grp = f.create_group(name)
        grp.create_dataset('weights',data=net.params[name][0].data)
        grp.create_dataset('bias',data=net.params[name][1].data)

f.close()`
2. load h5 weight by keras


`def load_weights(model, weights_path):
    f = h5py.File(weights_path, 'r')
    for layer in model.layers:
        if layer.name[:3] == 'res':
            layer.set_weights([f[layer.name]['weights'][:], f[layer.name]['bias'][:]])
        elif layer.name[:2] == 'bn':
            scale_name = 'scale'+layer.name[2:]
            weights = []
            weights.append(f[scale_name]['weights'][:])
            weights.append(f[scale_name]['bias'][:])
            weights.append(f[layer.name]['weights'][:])
            weights.append(f[layer.name]['bias'][:])
            layer.set_weights(weights)
    model.get_layer('conv1').set_weights([f['conv1']['weights'][:], f['conv1']['bias'][:]])
    model.get_layer('fc8-new').set_weights([f['fc8-new']['weights'][:].T, f['fc8-new']['bias'][:]])
    return model
model = get_resC3D() #keras ResNet C3D model architecture, similar to prototxt in caffe
model = load_weights(model, 'c3d_resnet50-1.h5')
for layer in model.layers:
   if layer.name == 'conv1' or layer.name[:3] == 'res':
       original_w = K.get_value(layer.W)
       converted_w = convert_kernel(original_w)
       K.set_value(layer.W, converted_w)`
